editor/scene/2d/tiles/tile_map_layer_editor.cpp

```diff
@@ -3465,21 +3466,38 @@ void TileMapLayerEditorTerrainsPlugin::edit(ObjectID p_edited_tile_map_layer_id)
 	}
 }
 
+void TileMapLayerEditorTerrainsPlugin::update_layout(EditorDock::DockLayout p_layout) {
+	bool is_vertical = (p_layout == EditorDock::DockLayout::DOCK_LAYOUT_VERTICAL);
+	// Main Panel.
+	main_box_container->set_vertical(is_vertical);
+	tilemap_tab_terrains->move_child(terrains_tree, is_vertical ? 1 : 0);
+	tilemap_tab_terrains->set_vertical(is_vertical);
+
+	// Toolbar.
+	tilemap_tiles_tools_buttons->set_vertical(is_vertical);
+	tools_settings->set_vertical(is_vertical);
+	tools_settings_vsep->set_vertical(is_vertical);
+}
+
 TileMapLayerEditorTerrainsPlugin::TileMapLayerEditorTerrainsPlugin() {
-	main_vbox_container = memnew(VBoxContainer);
+	wide_toolbar = memnew(HBoxContainer);
+	main_box_container = memnew(BoxContainer);
+	main_box_container->set_vertical(true);
 	// FIXME: This can trigger theme updates when the nodes that we want to update are not yet available.
 	// The toolbar should be extracted to a dedicated control and theme updates should be handled through
 	// the notification.
-	main_vbox_container->connect(SceneStringName(theme_changed), callable_mp(this, &TileMapLayerEditorTerrainsPlugin::_update_theme));
-	main_vbox_container->set_name(TTR("Terrains"));
+	main_box_container->connect(SceneStringName(theme_changed), callable_mp(this, &TileMapLayerEditorTerrainsPlugin::_update_theme));
+	main_box_container->set_name(TTR("Terrains"));
```

Comment:
If you are changing this line anyway...
```suggestion
	main_box_container->set_name(TTRC("Terrains"));
```